keps/sig-api-machinery/3352-aggregated-discovery/README.md

```diff
@@ -641,6 +645,31 @@ feature and upgrade/downgrade is not a problem.
 
 ### Version Skew Strategy
 
+When moving from beta to GA, we will introduce a new API group version `apidiscovery.k8s.io/v2`.
+
+All clients v1.26 to v1.29 will only request for the beta API group version `apidiscovery.k8s.io/v1beta2`.
+
+To accommodate skew between the client and server (older client and newer server), the server will serve both v2 and v2beta1 versions based on the client request headers. The API server will continue to support v2beta1 until its removal in Kubernetes v1.33.
+
+To accommodate skew between an older server and newer client, starting in v1.30,
+client-go will request for both v2 and v2beta1 by sending a list of group versions
+requested (in order from v2, v2beta1, unaggregated) and the server will return the
+first group version that matches. Concretely, this is done using `Accept` headers with a single request.
+
+```
+Accept: application/json;as=APIGroupDiscoveryList;v=v2;g=apidiscovery.k8s.io,application/json;as=APIGroupDiscoveryList;v=v2beta1;g=apidiscovery.k8s.io,application/json
+```
+
+In the case of older servers, the server will only
+be able to match v2beta1. The client will support both v2 and v2beta1. This allows a
+newer client to communicate with an older server that supports only the beta version.
+Other clients should follow the same convention to support version skew, though a client
+that is only capable of processing v2 is sufficient if it only communicates with v1.30+ servers.
+Otherwise, the client will need to be ready to tolerate a 406 Not Acceptable response and handle
+the error appropriately.
+
+If there is no skew and both server and client are v1.30+, clients will still request for v2 and v2beta1, and the server will match the first group version and return v2.
+
```

Comment:
A few comments to latter parts of the doc:
1. Are you planning to start the deprecation of v2beta1 with the introduction of v2? If so, it should be explained in the appropriate section below. 
2. Missing question `###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?` from the [template](https://github.com/kubernetes/enhancements/blob/master/keps/NNNN-kep-template/README.md).
3. Missing answer for `###### What are other known failure modes?` in the PRR questionnaire. 
4. Empty `Implementation History` section.

Reply:
Updated